go/vt/vtgate/planbuilder/physical/subquery_planning.go

```diff
@@ -82,6 +82,37 @@ func optimizeSubQuery(ctx *plancontext.PlanningContext, op *abstract.SubQuery) (
 	return outerOp, nil
 }
 
+// Remove the given subquery from any predicates and vindex predicates.
+//
+// If a subquery is merged into it's outer query, it needs to be removed from
+// the route as it can't be used as a routing predicate anymore.
+func RemoveExtractedSubqueryPredicate(route *Route, extractedSubquery *sqlparser.ExtractedSubquery) {
+	for _, vp := range route.VindexPreds {
+		idx := -1
+
+		for j, option := range vp.Options {
+			if sqlparser.EqualsExpr(option.Predicates[0], extractedSubquery.Original) {
+				idx = j
```

Comment:
I guess we could `break` here after we found one matching option?

Reply:
I am unsure whether the extracted subquery can match more than one vindex option or not. @harshit-gangal, do you know?

Otherwise, let's break yes

Reply:
I double checked this. Even if the same inner query exists multiple times in an outer query, each `ExtractedSubquery` object is unique. This is due to the unique name generated for each subquery's replacement bind variable.

I added a comment over at https://github.com/vitessio/vitess/pull/11104/files#diff-ecda8980a2e86d0f2a98ac67bb7b55d759fa7d1e8037dab2497f8fd167ada4c5R120-R121 that describes this.